populator-machinery/controller.go

```diff
@@ -206,6 +208,9 @@ func RunControllerWithConfig(vpcfg VolumePopulatorConfig) {
 	klog.Infof("Starting populator controller for %s", vpcfg.Gk)
 
 	stopCh := make(chan struct{})
+	if vpcfg.StopCh != nil {
```

Comment:
Can we add a unit test for this functionality?

Reply:
We can't not test the RunControllerWithConfig with fake kubeClient etc, if we want to test this function we might need to fake the controller? But that will not be test the really controller logic? I feel an e2e test might be more appropriate? Or is there an example of how to test this?

Reply:
Yes, create a new issue for it https://github.com/kubernetes-csi/lib-volume-populator/issues/166